app/controllers/saml_completion_controller.rb

```diff
@@ -26,11 +30,8 @@ def verify_sp_session_exists
     render_not_found unless sp_session.present? && sp_session[:request_url].present?
   end
 
-  def build_action_path(path)
-    path_year = path[-4..-1]
-    path = "/api/saml/finalauthpost#{path_year}"
-    recognized_path = Rails.application.routes.recognize_path(path, method: :post)
-
-    path if recognized_path[:controller] == 'saml_idp' && recognized_path[:action] == 'auth'
+  def valid_path?(action_path)
+    route = Rails.application.routes.recognize_path(action_path, method: :post)
+    route[:action] != 'page_not_found'
```

Comment:
I sorta feel the previous checks were a bit more precise / strict in a good sense? Was there a reason we needed to make it broader?

```suggestion
    route[:controller] == 'saml_idp' && route[:action] == 'auth'
```

Reply:
The old check was a guard against building URLs as strings, this now uses the Rails URL methods which are fairly trustworthy. If we change the URL we use intentionally, the "not found" behavior stays the same, but we might have to change which controller we hardcore, so I think this retains safety overall and is more flexible

Reply:
I think I'd rather keep the strictness as well, I don't think the flexibility will be needed.

Reply:
SGTM, brought back the check in ae3b2f004e17f6c43077f4f17043567f8c9f18e6